### PR TITLE
modernize CI

### PR DESCRIPTION
**Pull request by dermetfan**
- update tullia
- build all available platforms
  - report one github status each

### CODE REVIEW

File: nix/tullia.nix [hunk 1]
```diff
@@ -14,29 +14,35 @@ the rule that describes when a Tullia task is to be invoked.
...
         };
       };
 
+      command.text = let
+        flakeUrl = ''github:${repository}/"$(${lib.escapeShellArg config.preset.github.status.revision})"'';
+      in config.preset.github.status.lib.reportBulk {
+        bulk.text = ''
+          echo '["x86_64-linux", "x86_64-darwin"]' | nix-systems -i \
+          | jq 'with_entries(.key |= {"x86_64-linux": "linux", "x86_64-darwin": "macos"}[.])'
+        '';
```
Comment (angerman) [inline nix/tullia.nix:39]: I _do_ think we want to retain the architecture? We'll have aarch64-darwin in the system soon, and we'll need to distinguish between both.

**Comment by angerman** [comment]
The broken GHA CI should be fixed in #4650 soon.

File: nix/tullia.nix (hunk 1 above)
Comment (dermetfan) [reply nix/tullia.nix:39]: At the moment there's only `.#hydraJobs.{linux,macos}`. I guess aarch64-darwin would be added as another key like `macos-m1`. We can update this later when that key is added.

IMHO we should ideally just have `.#hydraJobs.{x86_64-linux,x86_64-darwin,aarch64-darwin}` but that is out of scope of this PR.

**Comment by Jimbo4350** [comment]
@dermetfan can this be rebased and merged? #4650 has been merged

**Comment by dermetfan** [comment]
bors try

**Comment by dermetfan** [comment]
bors merge

**Comment by dermetfan** [comment]
bors merge

**Comment by dermetfan** [comment]
bors merge

**Comment by dermetfan** [comment]
bors try

**Pull request merged**